client/rest-high-level/src/main/java/org/elasticsearch/client/IngestClient.java

```diff
@@ -111,4 +113,30 @@ public void deletePipelineAsync(DeletePipelineRequest request, ActionListener<Wr
         restHighLevelClient.performRequestAsyncAndParseEntity( request, RequestConverters::deletePipeline,
             WritePipelineResponse::fromXContent, listener, emptySet(), headers);
     }
+
+    /**
+     * Simulate a pipeline on a set of documents provided in the request
+     * <p>
+     * See
+     * <a href="https://www.elastic.co/guide/en/elasticsearch/reference/master/simulate-pipeline-api.html">
+     *     Simulate Pipeline API on elastic.co</a>
+     */
+    public SimulatePipelineResponse simulatePipeline(SimulatePipelineRequest request, Header... headers) throws IOException {
```

Comment:
I see that the API has changed a bit and this is not deprecated. I will switch this to `RequestOptions` once a review of the other stuff has been made.

Reply:
it's in, you can do so ;) can you also improve the javadocs adding the `@param` etc. like in the other methods?